homeassistant/components/bang_olufsen/websocket.py

```diff
@@ -101,6 +105,16 @@ def on_active_listening_mode(self, notification: ListeningModeProps) -> None:
             notification,
         )
 
+    def on_button_notification(self, notification: ButtonEvent) -> None:
+        """Send button dispatch."""
+        assert notification.state
```

Comment:
`assert` shouldn't be used except for when circumventing the type checker I think. If the latter is the case here, the assert should be put inside `if TYPE_CHECKING:`.

Reply:
Yes. I've put it in a `if TYPE_CHECKING ` now.